lib/vanagon/component/dsl.rb

```diff
@@ -416,21 +416,68 @@ def environment(*env) # rubocop:disable Metrics/AbcSize
         eos
       end
 
+      # Checks that the array of pkg_state is valid (install AND/OR upgrade).
+      # Returns vanagon error if invalid
+      #
+      # @param pkg_state [Array] array of pkg_state input to test
+      def check_pkg_state_array(pkg_state)
+        if pkg_state.empty? || (pkg_state - ["install", "upgrade"]).any?
+          raise Vanagon::Error, "#{pkg_state} should be an array containing one or more of ['install', 'upgrade']"
+        end
+      end
+
       # Adds action to run during the preinstall phase of packaging
       #
-      # @param pkg_state [Array] the state in which the scripts should execute. Can be
-      #   one or multiple of 'install' and 'upgrade'.
+      # @param pkg_state [String, Array] the package state during which the scripts should execute.
+      #   Accepts either a single string ("install" or "upgrade"), or an Array of Strings (["install", "upgrade"]).
       # @param scripts [Array] the Bourne shell compatible scriptlet(s) to execute
       def add_preinstall_action(pkg_state, scripts)
         pkg_state = Array(pkg_state)
         scripts = Array(scripts)
-
-        if pkg_state.empty? || !(pkg_state - ["install", "upgrade"]).empty?
-          raise Vanagon::Error, "#{pkg_state} should be an array containing one or more of ['install', 'upgrade']"
-        end
+        check_pkg_state_array(pkg_state)
         @component.preinstall_actions << OpenStruct.new(:pkg_state => pkg_state, :scripts => scripts)
       end
 
+      # Adds trigger for scripts to be run on specified pkg_state.
+      #
+      # @param pkg_state [String, Array] the package state during which the scripts should execute.
+      #   Accepts either a single string ("install" or "upgrade"), or an Array of Strings (["install", "upgrade"]).
+      # @param scripts [Array] the rpm pkg scriptlet(s) to execute
+      # @param pkg [String] the package the trigger will be set in
+      def add_install_rpm_triggers(pkg_state, scripts, pkg)
```

Comment:
I would call this `add_rpm_install_triggers` to be consistent with how you've named debian things below.

Reply:
+1